aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,126 @@
+
+* **Singleton** shares state across all users of the application and will lead to inappropriate concurrent use.
+* **Scoped** (the default) poses a similar issue between components for the same user.
+* **Transient** results in a new instance per request, but as components can be long-lived this results in a longer-lived context than may be intended.
+
+## Use of DbContext in Blazor Server Apps
+
+The following recommendations are designed to provide a consistent approach to using EF Core in Blazor Server apps. 
+
+* By default, consider using one context per operation. The context is designed for fast, low overhead instantiation:
+
+    ```csharp
+    var using context = new MyContext();
+    return await context.MyEntities.ToListAsync();
+    ```
+* Use a flag to prevent multiple concurrent operations:
+
+    ```csharp
+    if (Loading)
+    {
+        return;
+    }
+    try 
+    {
+        Loading = true;
+        // operations go here
+    }
+    finally 
+    {
+        Loading = false;
+    }
+    ```
+* For longer-lived operations that take advantage of EF Core's [change tracking](https://docs.microsoft.com/ef/core/querying/tracking) and/or [concurrency control](https://docs.microsoft.com/ef/core/saving/concurrency), scope the context to the lifetime of the component.
+
+### Create new DbContext instances
+
+The fastest way to create a new `DbContext` instance is by using `new` to create a new instance. There are several scenarios that may require additional dependencies to be resolved. For example, you may wish to use [DbContextOptions](https://docs.microsoft.com/ef/core/miscellaneous/configuring-dbcontext#configuring-dbcontextoptions) to configure the context. One approach is to register the options in DI and inject them into the component, like this:
+
+```razor
+@inject DbContextOptions<MyContext> Options
+
+@code {
+    public Task LoadAsync(int id)
+    {
+        using var context = new MyContext(Options);
+        return context.MyEntities.SingleOrDefaultAsync(e => e.Id == id);
+    }
+}
+```
+
+Another solution is to use a factory. EF Core 5.0 (preview) provides a built-in factory for creating new contexts. Because the sample app is based on the latest stable versions and not the preview, it implements its own factory in `Data/DbContextFactory.cs`. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
+
+The factory is injected into components and used to create new instances. For example, in `Pages/Index.razor`:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor?range=199-212)]
```

Comment:
Should we show how to configure the factory as a service and how to inject it here? I know you can just look at the sample app, but it seems convenient to point out all of the specific code involved.

Reply:
I included that earlier in the document.